### PR TITLE
LB-567: Merge test scripts

### PR DESCRIPTION
**Pull request by shivam-kapila**
# Problem

Listenbrainz currently has 4 test files
- _test.sh_
- _integration-test.sh_
- _frontend-test.sh_
- _spark_test.sh_

These all can be merged into one file.

# Solution

All these can be merged into `test.sh` and appropriate options can be provided to run particular tests.

- `test.sh` (for all tests)
- `test.sh fe` (for frontend tests)
- `test.sh spark` (for spark tests)
- `test.sh int` (for integration tests)

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -1,20 +1,33 @@
...
-# ./test.sh [params]       run tests, passing optional params to inner test
-# ./test.sh -s             stop test containers without removing
-# ./test.sh -d             clean test containers
+# ./test.sh -u             build unit test containers, bring up background and load database if needed
+# ./test.sh [params]       run unit tests, passing optional params to inner test
+# ./test.sh -s             stop unit test containers without removing
+# ./test.sh -d             clean unit test containers
+
+# FRONTEND TESTS
+# ./test.sh frontend       run frontend tests
```
Comment (paramsingh) [inline test.sh:12]: Can we call this `./test.sh -fe`, I don't wanna type `frontend` everytime

File: test.sh [hunk 2]
```diff
@@ -1,20 +1,33 @@
...
+# ./test.sh frontend -u    run frontend tests, update snapshots
```
Comment (paramsingh) [inline test.sh:13]: does `fe -u` do two things?

File: test.sh (hunk 1 above)
Comment (shivam-kapila) [reply test.sh:12]: Sure. Thats too long :p

File: test.sh (hunk 2 above)
Comment (shivam-kapila) [reply test.sh:13]: Oh no thats a mistake. Sorry

**Comment by shivam-kapila** [comment]
@brainzbot retest this please

File: test.sh [hunk 3]
```diff
@@ -65,68 +88,215 @@ function is_db_exists {
...
+    unit_dcdown
 else
     # Else, we have containers, just run tests
+    echo "Running tests"
     docker-compose -f $COMPOSE_FILE_LOC \
                    -p $COMPOSE_PROJECT_NAME \
-                   run --rm listenbrainz py.test "$@"
-fi
+                run --rm listenbrainz py.test "$@"
+fi
```
Comment (vansika) [inline test.sh:302]: new line?

Comment (shivam-kapila) [reply test.sh]: Added

**Pull request merged**